### ISSUE TITLE
Style/RedundantBegin error when using "super do" form

### ISSUE THREAD

**Issue by ben** [rubocop-hq/rubocop#6277]
I'm seeing a Rubocop error on code that Rubocop parsed just fine in 0.58.2.

## Expected behavior

Running rubocop 0.59.0 on this code:

```rb
class A
  def foo
    yield(1, 2, 3)
  end
end

class B < A
  def foo
    super do |x, y, z|
    end
  end
end
```

Should result in several recommendations, but no errors.

## Actual behavior

Here is the actual output:

```
➜ rubocop test.rb
Inspecting 1 file
An error occurred while Style/RedundantBegin cop was inspecting /private/tmp/test.rb:11:4.
To see the complete backtrace run rubocop -d.
C

Offenses:

test.rb:3:1: C: Style/Documentation: Missing top-level class documentation comment.
class A
^^^^^
test.rb:9:1: C: Style/Documentation: Missing top-level class documentation comment.
class B < A
^^^^^

1 file inspected, 2 offenses detected

1 error occurred:
An error occurred while Style/RedundantBegin cop was inspecting /private/tmp/test.rb:11:4.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop-hq/rubocop/issues

Mention the following information in the issue report:
0.59.0 (using Parser 2.5.1.2, running on ruby 2.5.0 x86_64-darwin17)
```

Here is the output from `rubocop -d`:

<details><pre><code>
➜ rubocop -d test.rb
For /private/tmp: configuration from /Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/config/default.yml
Inheriting configuration from /Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/config/enabled.yml
Inheriting configuration from /Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/config/disabled.yml
Inspecting 1 file
Scanning /private/tmp/test.rb
An error occurred while Style/RedundantBegin cop was inspecting /private/tmp/test.rb:11:4.
undefined method `stabby_lambda?' for s(:zsuper):RuboCop::AST::SuperNode
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/cop/style/redundant_begin.rb:68:in `on_block'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/cop/commissioner.rb:58:in `block (2 levels) in trigger_responding_cops'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/cop/commissioner.rb:106:in `with_cop_error_handling'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/cop/commissioner.rb:57:in `block in trigger_responding_cops'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/cop/commissioner.rb:56:in `each'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/cop/commissioner.rb:56:in `trigger_responding_cops'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/cop/commissioner.rb:34:in `block (2 levels) in <class:Commissioner>'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/ast/traversal.rb:100:in `on_def'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/cop/commissioner.rb:35:in `block (2 levels) in <class:Commissioner>'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/ast/traversal.rb:92:in `on_class'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/cop/commissioner.rb:35:in `block (2 levels) in <class:Commissioner>'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/ast/traversal.rb:50:in `block in on_begin'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/ast/traversal.rb:50:in `each'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/ast/traversal.rb:50:in `on_begin'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/cop/commissioner.rb:35:in `block (2 levels) in <class:Commissioner>'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/ast/traversal.rb:13:in `walk'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/cop/commissioner.rb:46:in `investigate'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/cop/team.rb:115:in `investigate'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/cop/team.rb:103:in `offenses'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/cop/team.rb:44:in `inspect_file'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/runner.rb:269:in `inspect_file'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/runner.rb:216:in `block in do_inspection_loop'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/runner.rb:248:in `block in iterate_until_no_changes'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/runner.rb:241:in `loop'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/runner.rb:241:in `iterate_until_no_changes'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/runner.rb:212:in `do_inspection_loop'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/runner.rb:115:in `block in file_offenses'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/runner.rb:133:in `file_offense_cache'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/runner.rb:113:in `file_offenses'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/runner.rb:101:in `process_file'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/runner.rb:78:in `block in each_inspected_file'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/runner.rb:75:in `each'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/runner.rb:75:in `reduce'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/runner.rb:75:in `each_inspected_file'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/runner.rb:67:in `inspect_files'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/runner.rb:39:in `run'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/cli.rb:157:in `execute_runner'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/cli.rb:85:in `execute_runners'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/lib/rubocop/cli.rb:41:in `run'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/exe/rubocop:13:in `block in <top (required)>'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/2.5.0/benchmark.rb:308:in `realtime'
/Users/bstraub/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.59.0/exe/rubocop:12:in `<top (required)>'
/Users/bstraub/bin/rubocop:23:in `load'
/Users/bstraub/bin/rubocop:23:in `<main>'
C

Offenses:

test.rb:3:1: C: Style/Documentation: Missing top-level class documentation comment.
class A
^^^^^
test.rb:9:1: C: Style/Documentation: Missing top-level class documentation comment.
class B < A
^^^^^

1 file inspected, 2 offenses detected

1 error occurred:
An error occurred while Style/RedundantBegin cop was inspecting /private/tmp/test.rb:11:4.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop-hq/rubocop/issues

Mention the following information in the issue report:
0.59.0 (using Parser 2.5.1.2, running on ruby 2.5.0 x86_64-darwin17)
Finished in 0.3956349999643862 seconds
</code></pre></details

## RuboCop version

```
0.59.0 (using Parser 2.5.1.2, running on ruby 2.5.0 x86_64-darwin17)
```

**Comment by nguyenductung** [comment]
This error has been fixed but not released yet. https://github.com/rubocop-hq/rubocop/issues/6261

**Comment by ben** [comment]
Great! I'll close this then.
